distribution/tools/plugin-cli/src/main/java/org/elasticsearch/plugins/InstallPluginCommand.java

```diff
@@ -253,6 +242,60 @@ private Path download(Terminal terminal, String pluginId, Path tmpDir) throws Ex
         return downloadZip(terminal, pluginId, tmpDir);
     }
 
+    // pkg private so tests can override
+    String getStagingHash() {
+        return System.getProperty(PROPERTY_STAGING_ID);
+    }
+
+    /** Returns the url for an official elasticsearch plugin. */
+    private String getElasticUrl(Terminal terminal, String stagingHash, Version version,
+                                        String pluginId, String platform) throws IOException {
+        final String baseUrl;
+        if (stagingHash != null) {
+            baseUrl = String.format(Locale.ROOT,
+                "https://staging.elastic.co/%1$s-%2$s/downloads/elasticsearch-plugins/%3$s", version, stagingHash, pluginId);
+        } else {
+            baseUrl = String.format(Locale.ROOT,
+                "https://artifacts.elastic.co/downloads/elasticsearch-plugins/%1$s", pluginId);
+        }
+        final String platformUrl = String.format(Locale.ROOT, "%1$s/%2$s-%3$s-%4$s.zip", baseUrl, pluginId, platform, version);
+        if (urlExists(terminal, platformUrl)) {
+            return platformUrl;
+        }
+        return String.format(Locale.ROOT, "%1$s/%2$s-%3$s.zip", baseUrl, pluginId, version);
```

Comment:
It looks like we do not need the argument index modifiers in any of the format strings in this method anymore?